addons/mail/controllers/discuss/channel.py

```diff
@@ -61,6 +69,15 @@ def _process_request_for_all(self, store: Store, name, params):
                 [("channel_id", "=", params["channel_id"]), ("is_self", "=", True)],
             ):
                 member.unpin_dt = False if params["pinned"] else fields.Datetime.now()
+            if (params["request_has_unpinned_channels"]):
+                # sudo: discuss.channel.member: sudo for performance. Checking existence of unpinned
+                # channels is acceptable, even if the channel is no longer accessible to the user.
+                store.add_global_values(
+                    has_unpinned_channels=request.env["discuss.channel.member"]
+                    .sudo()
+                    .search_count([("is_self", "=", True), ("is_pinned", "=", False)], limit=1)
+                    > 0,
+                )
```

Comment:
I would try to avoid duplicating its code though.

Method could have a variable that when set to true, it adds this value at the end.